pkg/alert/main.go

```diff
@@ -0,0 +1,122 @@
+package alert
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/iLert/ilert-go/v3"
+	shared "github.com/iLert/ilert-kube-agent"
+	v1 "github.com/iLert/ilert-kube-agent/pkg/apis/alert/v1"
+	agentclientset "github.com/iLert/ilert-kube-agent/pkg/client/clientset/versioned"
+	"github.com/iLert/ilert-kube-agent/pkg/config"
+	"github.com/iLert/ilert-kube-agent/pkg/utils"
+)
+
+var ilertClient *ilert.Client
+
+// CreateEvent creates an alert event
+func CreateEvent(
+	cfg *config.Config,
+	links []ilert.AlertLink,
+	alertKey string,
+	summary string,
+	details string,
+	eventType string,
+	priority string,
+	labels map[string]string,
+) error {
+	if ilertClient == nil {
+		ilertClient = ilert.NewClient(ilert.WithUserAgent(fmt.Sprintf("ilert-kube-agent/%s", shared.Version)))
+	}
+
+	if cfg.Settings.APIKey == "" {
+		log.Error().Msg("Failed to create an alert event. API key is required")
+		return errors.New("Failed to create an alert event. API key is required")
+	}
+
+	event := &ilert.Event{
+		AlertKey:  alertKey,
+		Summary:   summary,
+		Details:   details,
+		EventType: eventType,
+		APIKey:    cfg.Settings.APIKey,
+		Priority:  priority,
+		Links:     links,
+		CustomDetails: map[string]interface{}{
+			"labels": labels,
```

Comment:
@developer-nayminko we now support this as first party event field, please move labels to Labels: labels